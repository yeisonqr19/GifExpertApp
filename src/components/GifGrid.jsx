import React, { useState, useEffect } from "react";
import { getGifs } from "../helpers/GetGif";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then((gif) => setImages(gif));
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

export default GifGrid;
