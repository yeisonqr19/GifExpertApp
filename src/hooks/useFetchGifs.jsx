import { useState, useEffect } from "react";
import { getGifs } from "../helpers/GetGif";

export const useFetchGifs = (category) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((imgs) => {
      setstate({
        data: imgs,
        loading: false,
      });
    });
  }, [category]);

  return state; //este State me retorna Data:[] y loading en true
};
