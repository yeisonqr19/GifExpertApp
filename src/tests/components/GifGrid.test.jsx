import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas Para el componente GigGrid", () => {
  const category = "One piece ";

  test("El componente me debe renderizar y Hacer un SnapShot ", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de Mostrar Items Cuando se cargan imagenes con el <useFetchGif/>", () => {
    const gif = [
      {
        id: "abc",
        url: "https://localhost/images/imagen.jpg",
        title: "Titulo cualquiera",
      },
      {
        id: "123",
        url: "https://localhost/images/imagen.jpg",
        title: "Titulo cualquiera 2",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gif,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gif.length);
  });
});
