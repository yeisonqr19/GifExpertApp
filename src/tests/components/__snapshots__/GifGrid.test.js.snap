// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Para el componente GigGrid Debe de Mostrar Items Cuando se cargan imagenes con el <useFetchGif/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One piece 
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Titulo cualquiera"
      url="https://localhost/images/imagen.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Titulo cualquiera 2"
      url="https://localhost/images/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas Para el componente GigGrid El componente me debe renderizar y Hacer un SnapShot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One piece 
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
